paddle/fluid/pybind/sot/guards.h

```diff
@@ -236,6 +236,20 @@ class NumPyArrayValueMatchGuard : public GuardBase {
   PyObject* expected_;
 };
 
+class PyObjMatchGuard : public GuardBase {
```

Comment:
```suggestion
class WeakRefMatchGuard : public GuardBase {
```

就叫这个吧，和 `ValueMatchGuard` 作区分